File: cakk-client/src/main/java/com/cakk/client/vo/OidcPublicKeyList.java
package com.cakk.client.vo;

import static com.cakk.common.enums.ReturnCode.*;

import java.util.List;

import com.cakk.common.exception.CakkException;

public record OidcPublicKeyList(
	List<OidcPublicKey> keys
) {

	public OidcPublicKey getMatchedKey(String kid, String alg) {
		return keys.stream()
			.filter(key -> key.kid().equals(kid) && key.alg().equals(alg))
			.findAny()
			.orElseThrow(() -> new CakkException(EXTERNAL_SERVER_ERROR));
	}
}

File: cakk-external/src/main/kotlin/com/cakk/external/vo/key/OidcPublicKeyList.kt
package com.cakk.external.vo.key

import com.cakk.common.enums.ReturnCode
import com.cakk.common.exception.CakkException

@JvmRecord
data class OidcPublicKeyList(
	val keys: List<OidcPublicKey>
) {

	fun getMatchedKey(kid: String, alg: String): OidcPublicKey {
		return keys.find { key: OidcPublicKey ->
			key.kid == kid && key.alg == alg
		} ?: throw CakkException(ReturnCode.EXTERNAL_SERVER_ERROR)
	}
}
